drivers/display/display_smartbond.c

```diff
@@ -0,0 +1,634 @@
+		LOG_WRN("Could not switch to PLL");
+		return -EIO;
+	}
+#endif
+
+	return display_smartbond_configure(dev);
+}
+
+static void display_smartbond_dma_cb(const struct device *dma, void *arg,
+				uint32_t id, int status)
+{
+	struct display_smartbond_data *data = arg;
+
+	if (status < 0) {
+		LOG_WRN("DMA transfer did not complete");
+	}
+
+	k_sem_give(&data->dma_sync_sem);
+}
+
+static int display_smartbond_dma_config(const struct device *dev)
+{
+	struct display_smartbond_data *data = dev->data;
+
+	data->dma = DEVICE_DT_GET(DT_NODELABEL(dma));
+	if (!device_is_ready(data->dma)) {
+		LOG_ERR("DMA device is not ready");
+		return -ENODEV;
+	}
+
+	data->dma_cfg.channel_direction = MEMORY_TO_MEMORY;
+	data->dma_cfg.user_data = data;
+	data->dma_cfg.dma_callback = display_smartbond_dma_cb;
+	data->dma_cfg.block_count = 1;
+	data->dma_cfg.head_block = &data->dma_block_cfg;
+
+	/* Request an arbitrary DMA channel */
+	data->dma_channel = dma_request_channel(data->dma, NULL);
+	if (data->dma_channel < 0) {
+		LOG_ERR("Could not acquire a DMA channel");
+		return -EIO;
+	}
+
+	return 0;
+}
+
+
+static int display_smartbond_init(const struct device *dev)
+{
+	const struct display_smartbond_config *config = dev->config;
+	struct display_smartbond_data *data = dev->data;
+	int ret;
+
+	/* Device should be ready to be acquired */
+	k_sem_init(&data->device_sem, 1, 1);
+	/* Event should be signaled by LCDC ISR */
+	k_sem_init(&data->sync_sem, 0, 1);
+	/* Event should be signaled by DMA ISR */
+	k_sem_init(&data->dma_sync_sem, 0, 1);
+
+	/* As per docs, display port should be enabled by default. */
+	if (gpio_is_ready_dt(&config->disp)) {
+		ret = gpio_pin_configure_dt(&config->disp, GPIO_OUTPUT_ACTIVE);
+		if (ret < 0) {
+			LOG_ERR("Could not activate display port");
+			return -EIO;
+		}
+	}
+
+	data->current_pixel_size = DISPLAY_SMARTBOND_PIXEL_SIZE;
+	data->current_pixel_format = DT_INST_PROP(0, pixel_format);
+
+	ret = display_smartbond_resume(dev);
+	if (ret < 0) {
+		return ret;
+	}
+
+	ret = display_smartbond_dma_config(dev);
+	if (ret < 0) {
+		return ret;
+	}
+
+	IRQ_CONNECT(SMARTBOND_IRQN, SMARTBOND_IRQ_PRIO, smartbond_display_isr,
+								DEVICE_DT_INST_GET(0), 0);
+
+#if CONFIG_PM
+	/*
+	 * When in continues mode, the display device should always be refreshed
+	 * and so the controller is not allowed to be turned off. The latter, is
+	 * powered by PD_SYS which is turned off when the SoC enters the extended
+	 * sleep state. By acquiring PD_SYS, the deep sleep state is prevented
+	 * and the system enters the low-power state (i.e. ARM WFI) when possible.
+	 */
+	da1469x_pd_acquire(MCU_PD_DOMAIN_SYS);
+#endif
+
+	return 0;
+}
+
+static int display_smartbond_blanking_on(const struct device *dev)
+{
+	const struct display_smartbond_config *config = dev->config;
+	struct display_smartbond_data *data = dev->data;
+	int ret = 0;
+
+	k_sem_take(&data->device_sem, K_FOREVER);
+
+	/* XXX: Unsure if this feature is available in LCDC IP */
```

Comment:
Can this comment be updated? I'd prefer if this at least highlighted what the bit was doing, and that is might not be present on all IP revisions (assuming I'm understanding the current comment right)

Reply:
Sure. This bit will force LCD controller's output to blank that is, the controller will keep operating without outputting any pixel data. Think is, that functionality is not present on all IP revisions and this is why I added some comments.